### DISCUSSION TITLE
incoming wormhole Alarm?

### DISCUSSION THREAD

**Discussion by Misfualt** [Povstalec/StargateJourney#120] [Q&A] [answered]
Does any know of a program you could use to add an alarm using the speaker in CC: tweaked. If so if anyone could send the code that would be amazing. I've tried to but I'm just getting random noises.

**Comment by Povstalec** [comment, marked as answer]
Here's some old testing code I used some time back.
```
function incomingWormholeMonitoring()
	  
    while true
    do
        local event, address = os.pullEvent("stargate_incoming_wormhole")
        
		if #address > 0 then
		
			for symbol = 1,#address,1
			do
				print("Symbol ",symbol," = ", address[symbol])
			end
		
		end
		
        sleep(0)
    end
	
end
```
It should still work, just note the address will be nil unless you are using an Advanced Crystal Interface.
